File: kotlin/vci-client/src/main/java/io/mosip/vciclient/credentialResponse/CredentialResponseFactory.kt
package io.mosip.vciclient.credentialResponse

import io.mosip.vciclient.constants.CredentialFormat
import io.mosip.vciclient.credentialResponse.types.ldpVc.LdpVcCredentialResponseFactory
import io.mosip.vciclient.credentialResponse.types.msoMdocVc.MsoMdocVcCredentialResponseFactory

interface CredentialResponseFactory {
    companion object {
        fun createCredentialResponse(
            formatType: CredentialFormat,
            response: String,
        ): CredentialResponse {
            return when (formatType) {
                CredentialFormat.LDP_VC -> LdpVcCredentialResponseFactory().constructResponse(
                    response
                )

                CredentialFormat.MSO_MDOC -> MsoMdocVcCredentialResponseFactory().constructResponse(
                    response
                )
            }
        }
    }

    fun constructResponse(response: String): CredentialResponse
}

File: js/src/credentialResponse/CredentialResponseFactory.js
const {CredentialFormat} = require("../constants/CredentialFormat");
const DownloadFailedException = require("../exception/DownloadFailedException");
const LdpVcCredentialResponse = require("./types/ldpVc/LdpVcCredentialResponse");

class CredentialResponseFactory {
    static createCredentialResponse(formatType, response) {
        switch (formatType) {
            case CredentialFormat.LDP_VC:
                return new LdpVcCredentialResponse().toJson(response);
            default:
                const errorMessage = `Unsupported credential format type: ${formatType}`;
                Logger.error(errorMessage);
                throw new DownloadFailedException(errorMessage);
        }
    }
}

module.exports = CredentialResponseFactory;